File: update_phones_by_tc.mjs
import { createClient } from '@supabase/supabase-js';
import fs from 'fs';
import { parse } from 'csv-parse/sync';

// Read environment variables
const supabaseUrl = process.env.VITE_SUPABASE_URL;
const supabaseKey = process.env.VITE_SUPABASE_SERVICE_ROLE_KEY || process.env.VITE_SUPABASE_ANON_KEY;

if (!supabaseUrl || !supabaseKey) {
  console.error('Error: Missing Supabase credentials');
  process.exit(1);
}

// Initialize Supabase client
const supabase = createClient(supabaseUrl, supabaseKey);

// Read and parse CSV (semicolon-separated)
const csvContent = fs.readFileSync('Kitap1.csv', 'utf-8');
const records = parse(csvContent, {
  columns: true,
  skip_empty_lines: true,
  trim: true,
  delimiter: ';'
});

// Build TC to Phone mapping
const tcPhoneMap = new Map();

for (const row of records) {
  const tcNo = (row['T.C. Kimlik No'] || row['TC_No'])?.trim();
  const phoneNo = (row['Telefon No'] || row['Telefon_No'])?.trim();

  // Skip invalid data
  if (!tcNo || !phoneNo || tcNo === 'T.C. Kimlik No') continue;

  // Skip scientific notation or too short numbers
  if (phoneNo.includes('e+') || phoneNo.includes('E+') || phoneNo.length < 10) {
    continue;
  }

  // Clean phone number
  let cleanPhone = phoneNo.replace(/[^\d]/g, '');

  // Turkish phone numbers should be 10 or 11 digits
  if (cleanPhone.length >= 10) {
    // If it starts with 0, keep it, otherwise add 0 prefix
    if (!cleanPhone.startsWith('0') && cleanPhone.length === 10) {
      cleanPhone = '0' + cleanPhone;
    }
    // Trim to 11 digits max
    cleanPhone = cleanPhone.substring(0, 11);

    tcPhoneMap.set(tcNo, cleanPhone);
  }
}

console.log(`Found ${tcPhoneMap.size} valid TC-Phone mappings\n`);

// Update members table
let updatedCount = 0;
let notFoundCount = 0;
let errorCount = 0;

for (const [tcNo, phoneNo] of tcPhoneMap.entries()) {
  try {
    // Check if member exists with this TC number
    const { data: members, error: selectError } = await supabase
      .from('members')
      .select('id, full_name, phone')
      .eq('tc_identity_no', tcNo);

    if (selectError) {
      throw selectError;
    }

    if (members && members.length > 0) {
      const member = members[0];
      const oldPhone = member.phone || 'N/A';

      // Only update if phone is different
      if (oldPhone !== phoneNo) {
        const { error: updateError } = await supabase
          .from('members')
          .update({ phone: phoneNo })
          .eq('tc_identity_no', tcNo);

        if (updateError) {
          throw updateError;
        }

        updatedCount++;
        console.log(`✓ Updated ${member.full_name} (TC: ${tcNo}): ${oldPhone} → ${phoneNo}`);
      }
    } else {
      notFoundCount++;
    }
  } catch (error) {
    errorCount++;
    console.error(`✗ Error updating TC ${tcNo}:`, error.message);
  }
}

console.log(`\n=== Summary ===`);
console.log(`Total TC-Phone mappings in CSV: ${tcPhoneMap.size}`);
console.log(`Successfully updated: ${updatedCount}`);
console.log(`Not found in database: ${notFoundCount}`);
console.log(`Errors: ${errorCount}`);

File: update_phones_by_tc.py
#!/usr/bin/env python3
import csv
import os
from supabase import create_client, Client

# Read Supabase credentials from environment
supabase_url = os.environ.get('VITE_SUPABASE_URL')
supabase_key = os.environ.get('VITE_SUPABASE_SERVICE_ROLE_KEY')

if not supabase_url or not supabase_key:
    print("Error: Missing Supabase credentials")
    exit(1)

# Initialize Supabase client
supabase: Client = create_client(supabase_url, supabase_key)

# Read CSV file
tc_phone_map = {}
with open('extracted_csv.csv', 'r', encoding='utf-8') as f:
    reader = csv.DictReader(f)
    for row in reader:
        tc_no = row.get('TC_No', '').strip()
        phone_no = row.get('Telefon_No', '').strip()

        # Skip header row and invalid data
        if tc_no == 'TC_No' or not tc_no or not phone_no:
            continue

        # Clean phone number (remove scientific notation issues)
        # If phone is in scientific notation like "5.32e+009", skip it
        if 'e+' in phone_no.lower() or len(phone_no) < 10:
            continue

        # Remove any non-digit characters except leading zeros
        clean_phone = ''.join(c for c in phone_no if c.isdigit())

        # Turkish phone numbers should be 10 or 11 digits
        if len(clean_phone) >= 10:
            # If it starts with 0, keep it, otherwise it's just the mobile part
            if not clean_phone.startswith('0') and len(clean_phone) == 10:
                clean_phone = '0' + clean_phone

            tc_phone_map[tc_no] = clean_phone

print(f"Found {len(tc_phone_map)} valid TC-Phone mappings")

# Update members table
updated_count = 0
not_found_count = 0
error_count = 0

for tc_no, phone_no in tc_phone_map.items():
    try:
        # Check if member exists with this TC number
        result = supabase.table('members').select('id, full_name, phone').eq('tc_identity_no', tc_no).execute()

        if result.data and len(result.data) > 0:
            member = result.data[0]
            old_phone = member.get('phone', 'N/A')

            # Update phone number
            update_result = supabase.table('members').update({
                'phone': phone_no
            }).eq('tc_identity_no', tc_no).execute()

            if update_result.data:
                updated_count += 1
                print(f"✓ Updated {member['full_name']} (TC: {tc_no}): {old_phone} → {phone_no}")
        else:
            not_found_count += 1

    except Exception as e:
        error_count += 1
        print(f"✗ Error updating TC {tc_no}: {str(e)}")

print(f"\n=== Summary ===")
print(f"Total TC-Phone mappings in CSV: {len(tc_phone_map)}")
print(f"Successfully updated: {updated_count}")
print(f"Not found in database: {not_found_count}")
print(f"Errors: {error_count}")
